docs/addons/jupyterhub.md

```diff
@@ -0,0 +1,66 @@
+
+For more information regarding a Jupyter notebook, please consult the [official documentation](https://jupyter-notebook.readthedocs.io/en/stable/notebook.html).
+
+## Usage
+
+```typescript
+import 'source-map-support/register';
+import * as cdk from 'aws-cdk-lib';
+import * as blueprints from '@aws-quickstart/eks-blueprints';
+
+const app = new cdk.App();
+
+const jupyterHubAddOn = new blueprints.addons.JupyterHubAddOn();
+const addOns: Array<blueprints.ClusterAddOn> = [ jupyterHubAddOn ];
+
+const blueprint = blueprints.EksBlueprint.builder()
+  .addOns(...addOns)
+  .build(app, 'my-stack-name');
+```
+
+To validate that the JupyterHub add-on is running ensure that the add-on deployments for the controller and the webhook are in `RUNNING` state:
+
+```bash
+# Assuming add-on is installed in the jupyterhub.
+$ kubectl get po -n jupyterhub
+NAME                              READY   STATUS    RESTARTS   AGE
+continuous-image-puller-2skff     1/1     Running   0          23m
+continuous-image-puller-m6s5f     1/1     Running   0          23m
+hub-7dc8888f68-67hnl              1/1     Running   0          23m
+jupyter-admin                     1/1     Running   0          16m
+proxy-5df778944c-brrbf            1/1     Running   0          23m
+user-scheduler-7dbd789bc4-8zrjs   1/1     Running   0          23m
+user-scheduler-7dbd789bc4-gcb8z   1/1     Running   0          23m
+```
+
+## Functionality
+
+1. Deploys the jupyterhub helm chart in `jupyterhub` namespace by default.
+2. Supports [standard helm configuration options](./index.md#standard-helm-add-on-configuration-options).
+
+***Note***: For custom helm values, please consult the [official documentation](https://zero-to-jupyterhub.readthedocs.io/en/latest/resources/reference.html#). 
+
+## Using JupyterHub
+
+JupyterHub, by default, creates a proxy that is exposed to a `LoadBalancer` type Kubernetes service, which will integrate with AWS Load Balancer as indicated when running the following command:
+
+```bash
+kubectl get svc -n jupyterhub
+NAME           TYPE           CLUSTER-IP      EXTERNAL-IP                                                               PORT(S)        AGE
+hub            ClusterIP      172.20.171.28   <none>                                                                    8081/TCP       26m
+proxy-api      ClusterIP      172.20.31.32    <none>                                                                    8001/TCP       26m
+proxy-public   LoadBalancer   172.20.14.210   xxxxxxxx-1234567890.us-west-2.elb.amazonaws.com   80:32733/TCP   26m
+```
+
+You can log into the JupyterHub portal by accessing the Load Balancer endpoint in any browser. 
+
+![JupyterHub Login](./../assets/images/jupyterhub-login-page.png)
+
+A default `admin` username with `admin` as password is created initially. Once logged in, you should be able to access the main portal page.
```

Comment:
Do we provide a way to supply a password (e.g. via secrets store add-on) so that this could be leveraged for production use cases?

Reply:
That is probably better for a patterns - I talked to Jake, and the best course of action is to keep this as simple, and specify the custom configurations as part of an overall pattern. This applies to all other configurations as you noted below. Thoughts?

Reply:
As long as there is a way to use this pattern with no manual intervention afterwards we should good. The docs can be updated to show how to pass the password, ideally aligning with our security best practices.

Reply:
So it sounds like we can supply a password as shown [here](https://zero-to-jupyterhub.readthedocs.io/en/latest/administrator/authentication.html#allowed-users-admin-users), but it's not meant for any production use. 

In the end we probably need to introduce other ways of authentication (i.e. OAuth or Cognito as stated below). I can  either:

1/ include this as a way to provide the password (using secrets) and alert that this is not good for production use case OR
2/ Provide OAuth or other authentication method

Reply:
oauth works, I assume it should be good for cognito or any other oidc/oauth2 provider.

Reply:
done. Left a doc link with examples of various IdP set up, as the code uses a generic oauth2 authenticator class within JupyterHub. Will create an issue for pattern with Cognito. It may require some code change, which will be a fast follow if needed.